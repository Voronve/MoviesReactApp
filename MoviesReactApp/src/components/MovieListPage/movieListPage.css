.movieListWrapper {
    background-color: #232323;
    padding: 0 50px;
}
.movieListWrapper hr {
    color: #080707;
}

.genreDashboard {
    display: flex;
    justify-content: space-between;
    border-bottom: 3px solid #424242;
}