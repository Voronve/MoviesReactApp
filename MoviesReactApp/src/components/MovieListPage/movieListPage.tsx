
import './movieListPage.css'
import React, { useState, useEffect } from 'react';
import { GenreSelect } from '../GenreSelect/genreSelect';
import { MovieList } from '../MovieList/movieList';
import { SortControl, sortValue } from '../SortControl/sortControl';
import config from '../../config.json';
import { fetchMoviesList } from '../../utils/axios';
import { useSearchParams, Outlet, useLocation } from 'react-router-dom';
import { SearchForm } from '../SearchForm/searchForm';
export interface MovieInfo {
    /** Movie id*/
    id: string;
    /** Image url*/
    poster_path: string;
    /**Movie title */
    title: string;
    /**Movie release year */
    release_date: number;
    /**Movie janres list */
    genres: string[];
    /**Movie vote_average */
    vote_average: number;
    /**Movie length */
    runtime: string;
    /**Movie overview */
    overview: string;
}
export interface SearchParams {
    searchQueryParam : 'string';
    sortCriterionParam : sortValue;
    activeGenreParam : 'string';
}

/** Movie list functionality section*/
export function MovieListPage() {
    let [searchParams, setSearchParams] = useSearchParams();

    let query = searchParams.get('query');
    let sortBy = searchParams.get('sortBy') as sortValue;
    let genre = searchParams.get('genre') || 'All';

    if(sortBy !== sortValue.date && sortBy !== sortValue.title) sortBy = sortValue.default;

    const [movieList, setMovieList] = useState([] as MovieInfo[]);

    const handleSortCriterionChange = (value: sortValue) => {
        sortBy = value;
        const params = { sortBy } as {query: string, sortBy: sortValue, genre: string};
        if(query) params.query = query;
        if(genre) params.genre = genre;

        setSearchParams(params);
    };

    const handleActiveGenreChange = (value: string) => {
        genre = value;
        const params = { genre } as {query: string, sortBy: sortValue, genre: string};
        if(query) params.query = query;
        if(sortBy) params.sortBy = sortBy;

        setSearchParams(params);
    };

    useEffect(() => {
        if(process.env.NODE_ENV !== 'test') {
            fetchMoviesList(sortBy || sortValue.default, query || '', genre || '', setMovieList);
        }
    },[sortBy, genre, query]);

    return (
        <div className="movieListWrapper">
            <Outlet/>
            <div className="genreDashboard">
                <GenreSelect selectedGenre={genre} onSelect={handleActiveGenreChange}/>
                <SortControl activeSorting={sortBy} sortBy={handleSortCriterionChange}/>
            </div>
            <MovieList list={movieList} />
        </div>
    );
}