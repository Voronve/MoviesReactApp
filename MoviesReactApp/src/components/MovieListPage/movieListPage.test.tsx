import { BrowserRouter, Route, Routes } from 'react-router-dom';
import { MovieListPage } from './movieListPage';
import { render } from '@testing-library/react';

describe("Testing MovieList page component", () => {

    test("Testing movie list page rendering ", () => {

        const { asFragment } = render(
        (<BrowserRouter>
            <Routes>
                <Route path='/' element={<MovieListPage/>}/>
            </Routes>
        </BrowserRouter>));

        expect(asFragment()).toMatchSnapshot();
    });
});