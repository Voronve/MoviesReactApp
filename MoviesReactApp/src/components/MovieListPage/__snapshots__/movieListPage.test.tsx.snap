// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing MovieList page component Testing movie list page rendering  1`] = `
<DocumentFragment>
  <div
    class="movieListWrapper"
  >
    <div
      class="genreDashboard"
    >
      <div
        class="genresListWrapper"
      >
        <ul
          class="genresList"
        >
          <li
            class="selected"
          >
            ALL
          </li>
          <li
            class=""
          >
            DRAMA
          </li>
          <li
            class=""
          >
            ROMANCE
          </li>
          <li
            class=""
          >
            ANIMATION
          </li>
          <li
            class=""
          >
            ADVENTURE
          </li>
          <li
            class=""
          >
            FAMILY
          </li>
          <li
            class=""
          >
            COMEDY
          </li>
          <li
            class=""
          >
            FANTASY
          </li>
          <li
            class=""
          >
            SCIENCE FICTION
          </li>
          <li
            class=""
          >
            ACTION
          </li>
        </ul>
      </div>
      <div
        class="sortBywrapper"
      >
        <nav>
          <div
            class="sortByText"
          >
            SORT BY
          </div>
          <ul
            class="dropdown-menu"
          >
            <li
              class=""
            >
              TITLE
            </li>
            <li
              class=""
            >
              RELEASE DATE
            </li>
          </ul>
          <div
            class="arrowDown"
          >
            â–¼
          </div>
        </nav>
      </div>
    </div>
    <div
      class="moviesList"
    />
  </div>
</DocumentFragment>
`;
